Source/JavaScriptCore/wasm/WasmTypeDefinition.h

```diff
@@ -120,9 +118,15 @@ class StructType {
     FieldType* storage(StructFieldCount i) { return i + m_payload; }
     const FieldType* storage(StructFieldCount i) const { return const_cast<StructType*>(this)->storage(i); }
 
+    size_t getFieldOffset(uint32_t fieldIndex) const { return m_fieldOffsets[fieldIndex]; }
+
+    size_t payloadSize() const { return m_payloadSize; }
+
 private:
     FieldType* m_payload;
     StructFieldCount m_fieldCount;
+    FixedVector<size_t> m_fieldOffsets;
```

Comment:
I think, we should validate that the total size of the struct does not exceed unsigned (if it exceeds, then anyway, it cannot be allocated in wasm due to 4GB memory), and use `unsigned` offset here.